tools/clang/test/SemaHLSL/hlsl/objects/HitObject/hitobject-in-buffer.hlsl

```diff
@@ -0,0 +1,4 @@
+// RUN: %dxc -T lib_6_9 %s -verify
+
+// expected-error@+1{{'dx::HitObject' is an object and cannot be used as a type parameter}}
+RWStructuredBuffer<dx::HitObject> InvalidBuffer;
```

Comment:
Should probably also block it from being used as a global constant buffer variable.

Reply:
Yes, but i see no infrastructure / pre-existing diagnostic in DXC to do this. I suggest we open an issue for this to address later.

Reply:
@tex3d - Tracking this change request in #7234 . Can we consider this non-blocking for this PR?

Reply:
Yes, sounds good.